In [1]:
import numpy as np
import pandas as pd
from datetime import datetime

In [8]:
ranking=pd.read_csv('../Datasets/FIFA/fifa_ranking-2022-10-13.csv', parse_dates=['rank_date'])

In [9]:
ranking.head()

rank country_full country_abrv  total_points  previous_points  rank_change  \
0     3      Germany          GER        2153.0           2162.0            0   
1    83     Honduras          HON        1194.0           1194.0            5   
2    82     Dominica          DMA        1199.0           1199.0            5   
3    81     Suriname          SUR        1200.0           1200.0            5   
4    80     Cameroon          CMR        1215.0           1215.0            6   

  confederation  rank_date  
0          UEFA 2003-07-16  
1      CONCACAF 2003-07-16  
2      CONCACAF 2003-07-16  
3      CONCACAF 2003-07-16  
4           CAF 2003-07-16

In [10]:
ranking.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10660 entries, 0 to 10659
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   rank             10660 non-null  int64         
 1   country_full     10660 non-null  object        
 2   country_abrv     10660 non-null  object        
 3   total_points     10660 non-null  float64       
 4   previous_points  10660 non-null  float64       
 5   rank_change      10660 non-null  int64         
 6   confederation    10660 non-null  object        
 7   rank_date        10660 non-null  datetime64[ns]
dtypes: datetime64[ns](1), float64(2), int64(2), object(3)
memory usage: 666.4+ KB


In [20]:
ranking_2022 = ranking[ranking['rank_date'].dt.strftime('%Y-%m') == '2022-10']

In [21]:
ranking_2022

rank         country_full country_abrv  total_points  previous_points  \
10473   174              Burundi          BDI        799.08           799.08   
10474   173  Antigua and Barbuda          ATG        804.42           804.42   
10475   172           Mozambique          MOZ        820.17           815.14   
10476   169              Lesotho          LES        847.17           842.00   
10477   170               Belize          BLZ        839.85           839.85   
...     ...                  ...          ...           ...              ...   
10655    57               Panama          PAN       1432.87          1425.65   
10656    58             Cameroon          CMR       1417.44          1417.44   
10657    59                Ghana          GHA       1416.77          1416.77   
10658    61                India          IND       1405.79          1425.51   
10659   167       Cayman Islands          CAY        854.05           854.05   

       rank_change confederation  rank_date  
10473            1           CAF 2022-10-13  
10474            1      CONCACAF 2022-10-13  
10475            2           CAF 2022-10-13  
10476            2           CAF 2022-10-13  
10477            2      CONCACAF 2022-10-13  
...            ...           ...        ...  
10655            0      CONCACAF 2022-10-13  
10656           -1           CAF 2022-10-13  
10657           -1           CAF 2022-10-13  
10658            3           AFC 2022-10-13  
10659            2      CONCACAF 2022-10-13  

[187 rows x 8 columns]

So there were 187 countries ranked by October 2022.  32 of which made it to the FIFA world cup.

In [25]:
ranking_2022[ranking_2022['rank']<33]

rank         country_full country_abrv  total_points  previous_points  \
10495     1                  USA          USA       2087.48          2111.47   
10579    23             Portugal          POR       1742.25          1688.06   
10580    24  Republic of Ireland          IRL       1739.46          1695.26   
10581    25             Scotland          SCO       1727.97          1739.83   
10582    26               Russia          RUS       1716.57          1717.04   
10583    27             Colombia          COL       1706.46          1700.40   
10584    28       Czech Republic          CZE       1689.06          1675.21   
10585    29            Argentina          ARG       1660.37          1660.37   
10586    22          New Zealand          NZL       1743.72          1741.94   
10587    30              Finland          FIN       1655.04          1669.24   
10588    32               Poland          POL       1652.49          1647.92   
10595    31                Wales          WAL       1652.67          1665.24   
10596    21          Switzerland          SUI       1777.65          1775.36   
10597    20              Belgium          BEL       1791.37          1823.83   
10598    19              Austria          AUT       1793.68          1813.37   
10599     2               Sweden          SWE       2066.30          2050.18   
10600     3              Germany          GER       2064.75          2059.75   
10601     4              England          ENG       2051.61          2039.44   
10602     5               France          FRA       2023.09          2037.69   
10603     6                Spain          ESP       1997.74          1983.13   
10604     7               Canada          CAN       1997.31          1986.51   
10605     8          Netherlands          NED       1994.69          2002.99   
10606     9               Brazil          BRA       1989.98          1975.82   
10607    10            Korea DPR          PRK       1940.00          1940.00   
10608    11                Japan          JPN       1918.16          1917.07   
10609    12               Norway          NOR       1910.78          1889.96   
10610    13            Australia          AUS       1903.10          1906.20   
10611    14                Italy          ITA       1865.09          1868.29   
10612    15             China PR          CHN       1862.87          1862.87   
10613    16              Iceland          ISL       1850.91          1870.66   
10614    17       Korea Republic          KOR       1849.16          1850.09   
10615    18              Denmark          DEN       1847.17          1855.49   

       rank_change confederation  rank_date  
10495            0      CONCACAF 2022-10-13  
10579           -4          UEFA 2022-10-13  
10580           -2          UEFA 2022-10-13  
10581            2          UEFA 2022-10-13  
10582            2          UEFA 2022-10-13  
10583            2      CONMEBOL 2022-10-13  
10584            0          UEFA 2022-10-13  
10585           -2      CONMEBOL 2022-10-13  
10586            0           OFC 2022-10-13  
10587            1          UEFA 2022-10-13  
10588            0          UEFA 2022-10-13  
10595            1          UEFA 2022-10-13  
10596            0          UEFA 2022-10-13  
10597            1          UEFA 2022-10-13  
10598           -1          UEFA 2022-10-13  
10599           -1          UEFA 2022-10-13  
10600            1          UEFA 2022-10-13  
10601            0          UEFA 2022-10-13  
10602            0          UEFA 2022-10-13  
10603           -2          UEFA 2022-10-13  
10604            0      CONCACAF 2022-10-13  
10605            2          UEFA 2022-10-13  
10606            0      CONMEBOL 2022-10-13  
10607            0           AFC 2022-10-13  
10608            0           AFC 2022-10-13  
10609           -1          UEFA 2022-10-13  
10610            1           AFC 2022-10-13  
10611           -1          UEFA 2022-10-13  
10612           -1           AFC

The allocation of slots for each confederation was confirmed by the FIFA Council on 25 December 2020. The slots for the two host nations were taken directly from the quotas allocated to their confederations.

* AFC (Asia): 6 slots (including co-hosts Australia)
* CAF (Africa): 4 slots
* CONCACAF (North America, Central America and the Caribbean): 4 slots
* CONMEBOL (South America): 3 slots
* OFC (Oceania): 1 slot (including co-hosts New Zealand)
* UEFA (Europe): 11 slots
* Inter-confederation play-off tournament: 3 slots

A ten-team play-off tournament will decide the final three spots at the Women's World Cup. The play-off slot allocation is as follows:

* AFC (Asia): 2 slots
* CAF (Africa): 2 slots
* CONCACAF (North America, Central America and the Caribbean): 2 slots
* CONMEBOL (South America): 2 slots
* OFC (Oceania): 1 slot
* UEFA (Europe): 1 slot

So it's not just the top 32 teams